src/java.base/share/classes/java/lang/String.java

```diff
@@ -1334,13 +1423,21 @@ private static byte[] encodeUTF8(byte coder, byte[] val, boolean doReplace) {
         return Arrays.copyOf(dst, dp);
     }
 
-    private static byte[] encodeUTF8_UTF16(byte[] val, boolean doReplace) {
+    /**
+     * @param exceptionClass The exception class where any non-null value
+     *                       indicates malformed or unmappable bytes will
+     *                       result in an exception to be thrown instead of
+     *                       getting replaced.
+     * @param <E> The exception type parameter to enable callers to avoid
+     *           having to declare the exception
+     */
+    private static <E extends Exception> byte[] encodeUTF8_UTF16(byte[] val, Class<E> exceptionClass) throws E {
```

Comment:
Would it be possible to add a brief method description to this method? It's the equivalent of decoding val with the UTF-8 charset, then encoding the result with the UTF-16 charset, right?

In passing, if rename exceptionClass to exClass then you'll get get some back some of the horizontal space in several areas.

Reply:
Pushed following changes:

- dae04e9e35e Improve Javadoc of touched `encode*()` methods
- b120d14fd96 Rename `exceptionClass` to `exClass`